bucket/drivers/s3/s3.go

```diff
@@ -0,0 +1,113 @@
+package s3
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/sirupsen/logrus"
+)
+
+type S3Driver struct {
+	//svc is IAM S3 service client
+	svc    *s3.S3
+	config *aws.Config
+}
+
+func New(config *aws.Config) *S3Driver {
+	sess, _ := session.NewSession(config)
+	// Create S3 service client
+	svc := s3.New(sess)
+	return &S3Driver{
+		svc:    svc,
+		config: config,
+	}
+}
+
+func (d *S3Driver) CreateBucket(name string) (string, error) {
+	_, err := d.svc.CreateBucket(&s3.CreateBucketInput{
+		Bucket: aws.String(name),
+		CreateBucketConfiguration: &s3.CreateBucketConfiguration{
+			LocationConstraint: aws.String(*d.config.Region),
+		},
+	})
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case s3.ErrCodeBucketAlreadyExists:
+				logrus.Error("Bucket name already in use!")
+				panic(err)
+			case s3.ErrCodeBucketAlreadyOwnedByYou:
+				logrus.Warn("Bucket exists and is owned by the requester")
+			default:
+				logrus.Error("Unknown error")
```

Comment:
Lets print the error here with a little more verbosity. Something like
`logrus.Errorf("An unknown error has occurred while creating bucket %s: %v", name, err)`